### PR TITLE
[Merged by Bors] - Update ballot id calculation

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
With the updated method to calculate ATX IDs the whole object including the signature is used for ID calculation. This PR updates the ID calculation of Ballots to align with this new method. See also https://github.com/spacemeshos/go-spacemesh/issues/3516

## Changes
Update method to calculate Ballot ID calculation; update existing tests to ensure Ballot ID can be calculated

## Test Plan
Existing tests still pass.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by fasmat** [comment]
bors try

File: go.mod [hunk 1]
```diff
@@ -27,11 +27,11 @@ require (
 	github.com/prometheus/common v0.37.0
 	github.com/pyroscope-io/pyroscope v0.28.1
 	github.com/seehuhn/mt19937 v1.0.0
-	github.com/spacemeshos/api/release/go v1.4.1-0.20220905145225-2baea06a0206
+	github.com/spacemeshos/api/release/go v0.0.0-20220909175001-1907c2fe076e
```
Comment (dshulyak) [inline go.mod]: why this is updated in this change

**Review by dshulyak — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**